src/apirequest.ts

```diff
@@ -229,52 +229,79 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
   if (parameters.mediaUrl && media.body) {
     options.url = parameters.mediaUrl;
     if (resource) {
-      // gaxios doesn't support multipart/related uploads, so it has to
-      // be implemented here.
-      params.uploadType = 'multipart';
-      const multipart = [
-        {'Content-Type': 'application/json', body: JSON.stringify(resource)},
-        {
-          'Content-Type':
-            media.mimeType || (resource && resource.mimeType) || defaultMime,
-          body: media.body, // can be a readable stream or raw string!
-        },
-      ];
-      const boundary = uuid.v4();
-      const finale = `--${boundary}--`;
-      const rStream = new stream.PassThrough({
-        flush(callback) {
-          this.push('\r\n');
-          this.push(finale);
-          callback();
-        },
-      });
-      const pStream = new ProgressStream();
-      const isStream = isReadableStream(multipart[1].body);
-      headers['Content-Type'] = `multipart/related; boundary=${boundary}`;
-      for (const part of multipart) {
-        const preamble = `--${boundary}\r\nContent-Type: ${part['Content-Type']}\r\n\r\n`;
-        rStream.push(preamble);
-        if (typeof part.body === 'string') {
-          rStream.push(part.body);
-          rStream.push('\r\n');
-        } else {
-          // Gaxios does not natively support onUploadProgress in node.js.
-          // Pipe through the pStream first to read the number of bytes read
-          // for the purpose of tracking progress.
-          pStream.on('progress', bytesRead => {
-            if (options.onUploadProgress) {
-              options.onUploadProgress({bytesRead});
-            }
-          });
-          part.body.pipe(pStream).pipe(rStream);
+      let multipart;
+      if (!isBrowser()) {
```

Comment:
I think to cleanup this function a bit, we should just split this into two methods, maybe:

```
browserMultipartUpload
```

and:

```
multipartUpload
```